content_scripts/VirtualCamera.js

```diff
@@ -300,6 +303,124 @@ import * as THREE from '../../thirdPartyCode/three/three.module.js';
+                }
+            }
+
+            // Handle two-finger drag to pan
+            const handlePan = (event) => {
+                event.preventDefault();
+                if (event.touches.length === 2) {
+                    analyzeTouchMovement(event); // pans because isStrafeRequested is true
+                }
+            }
+
+            // Handle pinch to zoom
+            const handlePinch = (event) => {
+                event.preventDefault();
+                if (event.touches.length === 2) {
+                    const touch1 = event.touches[0];
+                    const touch2 = event.touches[1];
+                    const currentDistance = Math.hypot(touch2.clientX - touch1.clientX, touch2.clientY - touch1.clientY);
+
+                    if (initialDistance === 0) { // indicates the start of the pinch gesture
+                        initialDistance = currentDistance;
+                        lastDistance = initialDistance;
+                    } else {
+                        // Calculate the pinch scale based on the change in distance over time.
+                        // 3 is empirically determined to feel natural. -= so bigger distance leads to closer zoom
+                        this.mouseInput.unprocessedScroll -= 3 * (currentDistance - lastDistance); 
+                        lastDistance = currentDistance;
+                    }
+                }
+            }
+
+            // Add multitouch event listeners to the document
+            document.addEventListener('touchstart',  (event) => {
+                if (!realityEditor.device.utilities.isEventHittingBackground(event)) return;
+
+                isMultitouchGestureActive = true;
+                didMoveAtAll = false;
+
+                if (event.touches.length === 1) {
+                    this.mouseInput.isRotateRequested = true; // rotate
+                    this.mouseInput.isStrafeRequested = false;
+                    this.mouseInput.last.x = 0;
+                    this.mouseInput.last.y = 0;
+                    // this.setFocusTargetCube(event);
+
+                } else if (event.touches.length === 2) {
+                    initialDistance = 0; // Reset pinch distance
+                    this.mouseInput.isRotateRequested = false;
+                    this.mouseInput.isStrafeRequested = true; // pan
+                    this.mouseInput.last.x = 0;
+                    this.mouseInput.last.y = 0;
+                }
+            });
+            document.addEventListener('touchmove',  (event) => {
+                if (!isMultitouchGestureActive) return;
+                event.preventDefault();
+
+                // Ensure regular zoom level
+                document.documentElement.style.zoom = '1';
+                // Ensure no page offset
+                window.scrollTo(0, 0);
+
+                if (event.touches.length === 1) {
+                    handleRotate(event);
+
+                } else if (event.touches.length === 2) {
```

Comment:
Good to know – I'll try updating to make it detect whether you are pinching or panning and do the appropriate action

Reply:
Actually seems to be working ok with panning & zooming at the same time, although I also implemented a version where it can only do one of them and it picks the best one. I'm going to leave it with doing both at once, for now, since it feels pretty natural. But if you see errors as a result, I can switch to the other version in future.